Assets/Scripts/MovingElements/EQ Visualization/SampleProcessor.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using UnityEngine.Profiling;
+
+public class SampleProcessor : MonoBehaviour
+{
+    [SerializeField]
+    MusicSampler Sampler;
+
+    public int NumBins;
+
+    int[] SamplesPerBin;
+    float[] Bins;
+
+    public Action Updated;
+
+    void Start()
+    {
+        Bins = new float[NumBins];
+        CalcSamplesPerBin();
+
+        Updated += () => { };
+    }
+
+    void OnEnable()
+    {
+        Sampler.Updated += SampleAndProcess;
+    }
+
+    void OnDisable()
+    {
+        Sampler.Updated -= SampleAndProcess;
+    }
+
+    void CalcSamplesPerBin()
+    {
+        SamplesPerBin = new int[NumBins];
+        for (int i = 0; i < SamplesPerBin.Length; i++)
+        {
+            //determine the number of samples to consolidate into this bin
+            int samplesPerBin = (int)(
+                Mathf.Pow(2, i) * Mathf.Pow(2, Sampler.GetArrayPow() - NumBins)
+            );
+
+            //each bin must have at least one sample
+            if (samplesPerBin < 1)
+            {
+                samplesPerBin = 1;
+            }
+            SamplesPerBin[i] = samplesPerBin;
+        }
+    }
+
+    public int GetNumSamplesAtIndex(int index)
+    {
+        if (index < 0 || index > SamplesPerBin.Length)
+        {
+            return 0;
+        }
+        return SamplesPerBin[index];
+    }
+
+    public void SampleAndProcess()
```

Comment:
why is this public?

Reply:
Leftover from old implementation, changed!